src/test/java/tetris/ActionTest.java

```diff
@@ -1,69 +1,107 @@
 package tetris;
 
-import org.junit.Assert;
 import org.junit.Test;
 
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings("deprecation")
 public class ActionTest {
     @Test
     public void testChecksoftDrop() {
         DummyController controller = new DummyController();
-        Grid grid = new Grid(controller, 2, 2);
-        DummyShapeO tetromino = new DummyShapeO(new Coordinate(0, 0));
+        Grid grid = new Grid(controller, 10, 20);
+        DummyShapeO tetromino = new DummyShapeO(new Coordinate(5, 0));
         Action.SOFT_DROP.attempt(tetromino, grid);
```

Comment:
I just run then in different situations, in one where the action is possible and in one where the action is not, where it therefore has to follow the second procedure. Maybe it's an idea to get the coordinates and see if their in the right place?